parser/src/lexer.rs

```diff
@@ -1105,6 +1105,84 @@ where
         let tok_end = self.get_pos();
         Ok((tok_start, ty, tok_end))
     }
+
+    fn lex_byte(&self, s: String) -> Result<Vec<u8>, LexicalError> {
+        let mut res = vec![];
+        let mut escape = false; //flag if previous was \
+        let mut hex_on = false; // hex mode on or off
+        let mut hex_value = String::new();
+
+        for c in s.chars() {
+            match c {
+                '\\' => {
+                    if escape {
+                        res.push(92);
+                        escape = false;
+                    } else {
+                        escape = true;
+                    }
+                }
+
+                'x' => {
+                    if escape {
+                        hex_on = true;
+                    } else {
+                        res.push(120);
+                    }
+                    escape = false;
+                }
+                't' => {
+                    if escape {
+                        res.push(9);
+                    } else {
+                        res.push(116);
+                    }
+                    escape = false;
+                }
+                'n' => {
+                    if escape {
+                        res.push(10);
+                    } else {
+                        res.push(110)
+                    }
+                    escape = false;
+                }
+                'r' => {
+                    if escape {
+                        res.push(13);
+                    } else {
+                        res.push(114)
+                    }
+                    escape = false;
+                }
+                x => {
+                    if hex_on {
```

Comment:
I think this if statement must be around the match like this:
```rust
if hex_on {
  process digit
} else {
   match c {
     // the rest
   }
}

Reply:
done, with a bigger match, i think, it'll be clearer